Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/aspect/EventAspectAdvice.java

```diff
@@ -176,52 +178,15 @@ public void logFailure(JoinPoint joinPoint, Throwable e) {
             builder.setThrowable(e);
             builder.setInvoker(sig.getDeclaringType().toString());
             builder.setMethod(sig.getName());
+            builder.setService(service);
+            builder.setVersion(version);
 
+            builder.buildMessageID();
+            builder.buildTransactionID();
+            builder.buildDescription();
 
-            EventDirector director = new EventDirector();
-            director.setEventBuilder(builder);
-            director.constructEvent();
-
-            eventRecorder.recordEvent(director.getEvent());
-
+            eventRecorder.recordEvent(builder.getEvent());
         }
-
-    }
-
-
-    @Autowired
-    public void setInboundMessageAdviceDelegate(EventAdviceDelegate inboundMessageAdviceDelegate) {
-        this.inboundMessageAdviceDelegate = inboundMessageAdviceDelegate;
-    }
-
-    @Autowired
-    public void setInboundProcessingAdviceDelegate(EventAdviceDelegate inboundProcessingAdviceDelegate) {
-        this.inboundProcessingAdviceDelegate = inboundProcessingAdviceDelegate;
-    }
-
-    @Autowired
-    public void setAdapterDelegationAdviceDelegate(EventAdviceDelegate adapterDelegationAdviceDelegate) {
-        this.adapterDelegationAdviceDelegate = adapterDelegationAdviceDelegate;
-    }
-
-    @Autowired
-    public void setOutboundMessageAdviceDelegate(EventAdviceDelegate outboundMessageAdviceDelegate) {
-        this.outboundMessageAdviceDelegate = outboundMessageAdviceDelegate;
-    }
-
-    @Autowired
-    public void setOutboundProcessingAdviceDelegate(EventAdviceDelegate outboundProcessingAdviceDelegate) {
-        this.outboundProcessingAdviceDelegate = outboundProcessingAdviceDelegate;
-    }
-
-    @Autowired
-    public void setNwhinInvocationAdviceDelegate(EventAdviceDelegate nwhinInvocationAdviceDelegate) {
-        this.nwhinInvocationAdviceDelegate = nwhinInvocationAdviceDelegate;
-    }
-
-    @Autowired
-    public void setEventRecorder(EventRecorder eventRecorder) {
-        this.eventRecorder = eventRecorder;
     }
 
 }
```

Comment:
I see the below unit tests were deleted but I don't see any testing added for the new logic.  Is this something that is still to be done or left out on purpose?

Reply:
There are new tests in EventAspectAdviceTest. The diff is large and not rendered by default on the Web viewer. I removed the SpringContext test because it did not work at all, and the tests have been commented out since 2013 if I recall. The AfterThrowing and BeforeAndAfter unit test classes were deleted mainly because the logic has been changed and the tests no longer apply, but there are new ones created in EventAspectAdviceTest with methods: verifyAnnotationThrowsException, testEventExceptionOverride, and testExceptionThrowing